code/datums/gamemode/objectives/objective.dm

```diff
@@ -42,7 +42,7 @@
 	if(M)
 		O.owner = M
 	if(F)
-		faction = F
+		O.faction = F
```

Comment:
I feel that the objective_holder's faction and mind variables should be removed entirely